slick/src/main/scala/slick/jdbc/JdbcDataSource.scala

```diff
@@ -14,12 +15,23 @@ import slick.SlickException
   * similar to a `javax.sql.DataSource` but simpler. Unlike [[JdbcBackend.DatabaseDef]] it is not a
   * part of the backend cake. This trait defines the SPI for 3rd-party connection pool support. */
 trait JdbcDataSource extends Closeable {
+
+  var pausable: Option[Pausable] = None
+
+  def registerPausable(p: Pausable) = {
+    pausable = Option(p)
```

Comment:
If this should only ever be assigned once, shouldn't registerPausable verify this invariant?